packages/frontend/apps/electron/src/main/protocol.ts

```diff
@@ -83,31 +83,56 @@ export function registerProtocol() {
     return handleFileRequest(request);
   });
 
-  // todo(@pengx17): remove this
   session.defaultSession.webRequest.onHeadersReceived(
     (responseDetails, callback) => {
       const { responseHeaders } = responseDetails;
-      if (responseHeaders) {
-        // replace SameSite=Lax with SameSite=None
-        const originalCookie =
-          responseHeaders['set-cookie'] || responseHeaders['Set-Cookie'];
-
-        if (originalCookie) {
-          delete responseHeaders['set-cookie'];
-          delete responseHeaders['Set-Cookie'];
-          responseHeaders['Set-Cookie'] = originalCookie.map(cookie => {
-            let newCookie = cookie.replace(/SameSite=Lax/gi, 'SameSite=None');
-
-            // if the cookie is not secure, set it to secure
-            if (!newCookie.includes('Secure')) {
-              newCookie = newCookie + '; Secure';
+      (async () => {
+        if (responseHeaders) {
+          const originalCookie =
+            responseHeaders['set-cookie'] || responseHeaders['Set-Cookie'];
+
+          if (originalCookie) {
+            // save the cookies, to support third party cookies
+            for (const cookies of originalCookie) {
+              const parsedCookies = cookieParser.parse(cookies);
+              for (const parsedCookie of parsedCookies) {
+                if (!parsedCookie.value) {
+                  await session.defaultSession.cookies.remove(
+                    responseDetails.url,
+                    parsedCookie.name
+                  );
+                } else {
+                  await session.defaultSession.cookies.set({
+                    url: responseDetails.url,
+                    domain: parsedCookie.domain,
+                    expirationDate: parsedCookie.expires?.getTime(),
+                    httpOnly: parsedCookie.httpOnly,
+                    secure: parsedCookie.secure,
+                    value: parsedCookie.value,
+                    name: parsedCookie.name,
+                    path: parsedCookie.path,
+                    sameSite: parsedCookie.sameSite?.toLowerCase() as
+                      | 'unspecified'
+                      | 'no_restriction'
+                      | 'lax'
+                      | 'strict'
+                      | undefined,
+                  });
+                }
+              }
             }
-            return newCookie;
-          });
-        }
-      }
+          }
 
-      callback({ responseHeaders });
+          responseHeaders['Access-Control-Allow-Origin'] = ['*'];
+          responseHeaders['Access-Control-Allow-Headers'] = ['*'];
```

Comment:
do we need this?

Reply:
This is to bypass CORS

Reply:
Need to consider the response header have lowercase header names
```
delete responseHeaders['access-control-allow-origin'];
delete responseHeaders['access-control-allow-headers'];
```